provider/win_driver_installer_test.py

```diff
@@ -563,3 +566,84 @@ def balloon_test(test, params, vm, balloon_test_win):
     error_context.context("Reset balloon memory...done", test.log.info)
     session.close()
     error_context.context("Balloon test done.", test.log.info)
+
+
+def get_keycode_cfg(filename):
+    """
+    Get keyname to keycode cfg table.
+    :param filename: filename that key to keycode file.
+    """
+    keycode_cfg_path = os.path.join(data_dir.get_deps_dir("key_keycode"), filename)
+    with open(keycode_cfg_path) as f:
+        return json.load(f)
+
+
+@error_context.context_aware
+def vioinput_test(test, params, vm):
+    """
+    Keyboard test. Support single key and combination key tests.
+
+    :param test: kvm test object
+    :param params: dictionary with the test parameters
+    :param vm: vm object
+    """
+
+    def key_check(key):
```

Comment:
This seems exactly same with other vioinput cases, shall we abstract it into another independent module like vioinput_basic, it is okay if there is only this single one function for it; and then all the other similar steps could call the basic function.

Reply:
Move functions inside vioinput_keyboard to provider and her in win_driver_installer_test we might have one simple `from vioinput_basic import key_tap_test as vioinput_test`

Reply:
Okay,sorry for the delay.Updated the code.Could you help to review it again?Thanks a lot!